### DISCUSSION TITLE
What is the RPN in mmdetection?

### DISCUSSION THREAD

**Discussion by yuhua666** [open-mmlab/mmdetection#9598] [General]
I hope to replicate the original version of fast rcnn. I noticed that the replicated version of fast rcnn of mmdetection includes Shared2FCBBoxHead, which is not what I want.
I see an RPN detector in mmdetection. It seems that there is a Shared2FCBBoxHead less than the fast rcnn version in the network structure. Is this the corresponding original fast rcnn?
In short, I hope to find a version of mmdetection similar to the original version of fast rcnn. Can anyone help me?

**Comment by Maenansyn** [comment]
Fast R-CNN takes an input image and region proposals as input, as described in the [original paper](https://arxiv.org/abs/1504.08083). So you need the RPN in front of Fast R-CNN as its output is part of the input of Fast R-CNN.. this is also noted in the respective [Git](https://github.com/open-mmlab/mmdetection/tree/master/configs/fast_rcnn#introduction).

The RPNHead is [here](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/dense_heads/rpn_head.py) and a default config as used in the later Faster R-CNN can be found in its [config](https://github.com/open-mmlab/mmdetection/blob/master/configs/_base_/models/faster_rcnn_r50_fpn.py#L19).
